### ISSUE TITLE
Couldn't open CUDA library libcuda.so.1 in when building using nvidia docker cuda8 image

### ISSUE THREAD

**Issue by gunan** [tensorflow/tensorflow#4931]
We built tensorflow on nvidia-docker image nvidia/cuda:8.0-cudnn5-devel
"""
However, when using the package we built, we run into this problem:
I tensorflow/stream_executor/dso_loader.cc:105] Couldn't open CUDA library libcuda.so.1. LD_LIBRARY_PATH: /usr/local/nvidia/lib:/usr/local/nvidia/lib64:
"""
In this special Dockerfile, we are using whatever LD_LIBRARY_PATH defined by nvidia docker.

It looks like we are looking specificly for the file libcuda.so.1
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/dso_loader.cc#L80

And it looks like we switched to doing this due to:
https://github.com/tensorflow/tensorflow/issues/2865

However, libcuda.so.1 is not linked/made available by nvidia-docker in the docker image.

Side note: it seems to be not available in our docker images based on 7.5, but that one somehow works.
No idea how.

@flx42 could you help us shed some light into this?

**Comment by gunan** [comment]
@yifeif is also working on this

**Comment by jart** [comment]
Is this related to #4397?

**Comment by flx42** [comment]
> However, libcuda.so.1 is not linked/made available by nvidia-docker in the docker image.

This file is only available at runtime when using `nvidia-docker`, if you don't see it, you have a problem.

**Comment by flx42** [comment]
@jart: no, I believe this is different. In #4397, the issue is that the install path of cuDNN varies depending on how it was installed (deb or tarball).

**Comment by gunan** [comment]
@flx42 When running nvidia-docker, I can verify that the following command does not return anything:
$ find / -name libcuda.so.1

However, in the host system I can verify that it is available.

**Comment by flx42** [comment]
@gunan please file a bug against `nvidia-docker`, and include the output of `ls -R /var/lib/nvidia-docker/volumes/nvidia_driver/`

This is what I have:
```
$ nvidia-docker run -ti nvidia/cuda sh -c 'find / -name libcuda.so.1'
/usr/local/nvidia/lib64/libcuda.so.1
/usr/local/nvidia/lib/libcuda.so.1
```

**Comment by gunan** [comment]
I got one machine working!
Was trying many different combinations, but finally using ubuntu 16.04, 361 driver from ubuntu repo, runfile installer to install just cuda toolkit, and nvidia cuda and cuda-devel image, and mucking with our Dockerfile, I got everything working.

I got rid of the old VM before this report, but I will create the issue once I can re-reproduce the failure.
Thanks for your input, and very quick response Felix!
